Libplanet/Net/UnboundPeer.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.Diagnostics.Contracts;
+using System.Net;
+using System.Runtime.Serialization;
+using Libplanet.Crypto;
+using Libplanet.Serialization;
+
+namespace Libplanet.Net
+{
+    [Serializable]
+    [Equals]
+    public class UnboundPeer : Peer
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="UnboundPeer"/> class.
+        /// </summary>
+        /// <param name="publicKey">A <see cref="PublicKey"/> of the
+        /// <see cref="Peer"/>.</param>
+        /// <param name="appProtocolVersion">An application protocol version
+        /// that the <see cref="Peer"/> is using.</param>
+        public UnboundPeer(
+            PublicKey publicKey,
+            int appProtocolVersion)
+            : this(publicKey, appProtocolVersion, null)
+        {
+        }
+
+        internal UnboundPeer(
+            PublicKey publicKey,
+            int appProtocolVersion,
+            IPAddress publicIPAddress)
+        : base(publicKey, null, appProtocolVersion, publicIPAddress)
+        {
+            PublicKey = publicKey ??
+                        throw new ArgumentNullException(nameof(publicKey));
+            EndPoint = null;
```

Comment:
IMHO we'd better get rid of this property from the abstract class `Peer`, and let the only `BoundPeer` have this property instead.